kamon-core/src/main/scala/kamon/trace/logging/MdcKeysSupport.scala

```diff
@@ -34,7 +39,12 @@ trait MdcKeysSupport {
 
   private[this] def copyToMdc(traceContext: TraceContext): Iterable[String] = traceContext match {
     case ctx: MetricsOnlyContext ⇒
-      ctx.traceLocalStorage.underlyingStorage.collect {
+
+      // Add the default key value pairs for the trace token and trace name.
```

Comment:
That is fine with me.

It does mean that calling `withMdc` has the side-effect of modifying TraceLocal's storage though. At the moment, it should return the `MDC` and TraceLocal storage to the initial state.